common/plugins/org.polarsys.capella.common.libraries/src/org/polarsys/capella/common/libraries/manager/LibraryManager.java

```diff
@@ -100,8 +112,14 @@ public void onLibraryProviderChanged(LibraryProviderEvent event) {
   @Override
   public Collection<IModel> getAllModels(TransactionalEditingDomain domain) {
     HashMap<IModelIdentifier, IModel> models = new HashMap<IModelIdentifier, IModel>();
+    List<AirdResource> airdResources = domain.getResourceSet().getResources().stream().filter(AirdResource.class::isInstance).map(AirdResource.class::cast).collect(Collectors.toList());
```

Comment:
the session lookup can be moved to CDOLibraryProvider. 

this code assumes that there is an aird and a session in the current domain, which may not be the case in this api

Reply:
The session handling here was just done so it could forward it to CDOLibraryProvider. Ok I get why you wanted a previous api with domain instead of session. I am updating it.